src/components/TemplatePopup.tsx

```diff
@@ -0,0 +1,220 @@
+          {`Create a ${templateType} from a template`}
+        </PopupHeader>
+        <InputBlock mb={'12px'}>
+          <Label>Type</Label>
+          <Select
+            onChange={(event) => {
+              setTemplateType(event.target.value)
+            }}
+            defaultValue="Script"
+            sx={{
+              border: "1px solid #C4C4C4",
+              fontSize: "14px",
+              color: "#000",
+              padding: "8px",
+              width: "100%",
+              fontWeight: "bold",
+              marginBottom: "5px",
+              borderRadius: "2px"
+            }}
+          >
+            <option>Script</option>
+            <option>Transaction</option>
+          </Select>
+        </InputBlock>
+        <InputBlock mb={'12px'}>
+          <Label>Capability</Label>
+          <Select 
+            onChange={(event) => {
+              setSelectedCapability(event.target.value)
+            }}
+            defaultValue={capabilitiesKeys[0]}
+            sx={{
+              border: "1px solid #C4C4C4",
+              fontSize: "14px",
+              color: "#000",
+              padding: "8px",
+              width: "100%",
+              fontWeight: "bold",
+              marginBottom: "5px",
+              borderRadius: "2px"
+            }}
+          >
+            {capabilitiesKeys.map((capabilityKey) => 
+              types[capabilityKey] === "Link" && 
+                <option 
+                  key={capabilityKey}
+                  value={capabilityKey}
+                >
+                  {`${capabilityKey} - get and borrow the Capability`}
+                </option>
+            )}
+          </Select>
+        </InputBlock>
+        <InputBlock mb={'24px'}>
+          <Label>Name</Label>
+          <Input
+            ref={firstInput}
+            value={templateName}
+            onChange={event => setTemplateName(event.target.value)}
+          />
+        </InputBlock>
+        <SpaceBetween>
+          <FlowButton className="grey modal" onClick={triggerClose}>
+            Close
+          </FlowButton>
+            <FlowButton
+              className="green modal"
+              onClick={async () => {
+                setProcessing(true);
+                const capData = capabilities[selectedCapability];
+
+                if (templateType=== "Transaction") {
+                    const res = await mutator.createTransactionTemplate(
+                      getInterpolatedTemplate(
+                        "tx", 
+                        capData.contractAddr, 
+                        capData.path, 
+                        `${capData.resourceContract}.${capData.resource}`, 
+                        capData.contractImplementedInterfaces.join(",")
+                      ), 
+                      templateName
+                    )
+                    navigate(`/${projectPath}?type=tx&id=${res.data?.createTransactionTemplate?.id}&storage=${selectedResourceAccount || 'none'}`)
```

Comment:
Same DRY for `${selectedResourceAccount || 'none'}`

Reply:
Changed in [this commit](https://github.com/onflow/flow-playground/pull/175/commits/289b5d04ba186213347bb27b87021fdfc8efaf3e)